src/Manager.php

```diff
@@ -30,6 +30,10 @@ public function __construct(StorageInterface $storage, RuleFactoryInterface $rul
 
     public function userHasPermission($userId, string $permissionName, array $parameters = []): bool
     {
+        if (!is_string($userId)) {
+            throw new \InvalidArgumentException(sprintf('userId must be a string, %s given.', gettype($userId)));
```

Comment:
No, that's expected that $userId can be integer. That is to improve usability when getting entity ID as int from database.